src/components/twilio-voice-dialer/public/twilio-voice-dialer.js

```diff
@@ -11,12 +11,35 @@ class TwilioVoiceDialer extends HTMLElement {
     this.attachShadow({ mode: 'open' });
   }
 
+  attributeChangedCallback() {
+    this.#render();
+  }
+
+  connectedCallback() {
+    this.shadowRoot
+      .querySelector('#call')
+      .addEventListener('click', () => this.#handleCall());
+    this.shadowRoot
+      .querySelector('#hangup')
+      .addEventListener('click', () => this.#handleHangup());
+  }
+
   disconnectedCallback() {
     this.#device.destroy();
   }
 
-  attributeChangedCallback() {
-    this.#render();
+  async #handleCall() {
+    const recipient = this.shadowRoot.querySelector('#recipient').value;
+    if (recipient) {
+      const call = await this.#device.connect({ params: { recipient } });
+      this.#setupCallHandlers(call);
+      this.#setStatus('inprogress');
```

Comment:
I know I recently just asked about using type annotations and we decided to decide at a later time, but seeing sentinel string values is worrying me. We could use a JS Record as an enumeration too, but using type annotations will at least give those using a language server type hinting and errors here if we don't call `setStatus` with the right values.

Reply:
I understand the concern here. If this is a big project and if we need to make these components consumable by customers, we would use TS instead. But since we're using vanilla JS and trying to make this as simple as possible. We should make it work without using third party libraries as much as possible.

Reply:
+1 for keeping as simple as possible